File: tests/eval-fixtures/abstract-domains/core/repo/domain.go
package abstractdomains

func unknownInt() int {
	return 7
}

func unsupported(any interface{}) interface{} {
	return any
}

func GoDomain(flag bool, input *int) string {
	// POLINT-FEATURE abstract-domains/go/initialized-locals
	var out string
	var count int
	// POLINT-FEATURE abstract-domains/go/string-constant
	out = "cold"
	// POLINT-FEATURE abstract-domains/go/nil-branch
	if input == nil {
		out = "nil"
	// POLINT-FEATURE abstract-domains/go/boolean-branch
	} else if flag {
		out = "warm"
		count = 1
	} else {
		out = "cool"
	}
	// POLINT-FEATURE abstract-domains/go/loop-widening
	for count < unknownInt() {
		count = count + 1
		if count > 2 {
			break
		}
	}
	// POLINT-FEATURE abstract-domains/go/unknown-call-havoc
	_ = unsupported(input)
	return out
}

File: tests/eval-fixtures/abstract-domains/core/repo/web/src/domain.ts
declare const external: (value: unknown) => unknown;

export function tsDomain(flag: boolean, input: string | null | undefined) {
  // POLINT-FEATURE abstract-domains/ts/maybe-uninitialized-local
  let label: string;
  let count = 0;
  // POLINT-FEATURE abstract-domains/ts/string-constant
  label = "cold";
  // POLINT-FEATURE abstract-domains/ts/nullish-branch
  if (input == null) {
    label = "missing";
  // POLINT-FEATURE abstract-domains/ts/boolean-branch
  } else if (flag) {
    label = "truthy";
  } else {
    label = "falsy";
  }
  while (count < 4) {
    count = count + 1;
  }
  const dynamicTarget: Record<string, unknown> = {};
  // POLINT-FEATURE abstract-domains/ts/dynamic-write-havoc
  dynamicTarget[String(input)] = external(label);
  // POLINT-FEATURE abstract-domains/ts/nullish-coalescing
  return { label, count, maybe: input ?? "fallback" };
}
